skimage/restoration/tests/test_j_invariant.py

```diff
@@ -27,10 +28,12 @@ def test_invariant_denoise():
     assert_(denoised_mse < original_mse)
 
 
-def test_invariant_denoise_color():
+@testing.parametrize('dtype', [np.float32, np.float64])
+def test_invariant_denoise_color(dtype):
     denoised_img_color = _invariant_denoise(
-            noisy_img_color, _denoise_wavelet,
-            denoiser_kwargs=dict(multichannel=True))
+        noisy_img_color.astype(dtype), _denoise_wavelet,
+        denoiser_kwargs=dict(multichannel=True))
+    assert denoised_img_color.dtype == dtype
```

Comment:
similar thought as above: won't this fail if `dtype` and `_float_type(dtype)` don't match?